src/common/decorators/swagger.ts

```diff
@@ -111,3 +115,26 @@ export function ApiTransactionResponse(
     })
   );
 }
+
+/**
+ * A helper that combines responses for SDK Error like `BadRequestException`, `NotFoundException`, `UnprocessableEntityException`, `InternalServerErrorException`
+ *
+ * @param options - these will be passed to the `ApiNotFoundResponse` decorator
+ */
+export function ApiTransactionFailedResponse(
+  options: ApiResponseOptions
+): ReturnType<typeof applyDecorators> {
+  return applyDecorators(
+    ApiNotFoundResponse(options),
+    ApiBadRequestResponse({
+      description: 'Returned if transaction validation failed or same data already exists in chain',
+    }),
+    ApiUnprocessableEntityResponse({
+      description:
+        'Returned if there is insufficient balance to perform action or required perquisites for transaction have not been met',
+    }),
+    ApiInternalServerErrorResponse({
```

Comment:
I don't think we should document anything in 500 range, since they are *supposed* to be unexpected.